### DISCUSSION TITLE
scope & need of enzyme adapter

### DISCUSSION THREAD

**Discussion by mtfoley** [open-sauced/open-sauced#1243] [Ideas]
The dependencies on `enzyme` and also the unofficial `enzyme` adapter appear to be indirectly used in these files:

https://github.com/open-sauced/open-sauced/blob/bf2e3c3c36f890b1ef132e13b4a0cf71789ca223/src/setupTestFramework.js
https://github.com/open-sauced/open-sauced/blob/93654fb65801283385cc423a3e04caced0353db0/src/tests/Dashboard.test.js
https://github.com/open-sauced/open-sauced/blob/93654fb65801283385cc423a3e04caced0353db0/src/tests/Repository.test.js

So am I interpreting correctly that the only use of enzyme is for the snapshot testing? And if so, what I would wonder about is the use case. If the snapshots don't move with the repo and only get generated on tests being run, would that not mean that these tests would only fail if the user followed these steps:
1) clone repo
2) npm ci
3) npm test
4) edit dashboard or repository components
5) tests fail

So all that said - do these tests make sense? If not, can we shed the enzyme dependency? I see jest and @testing-library in use elsewhere in the codebase.

**Comment by 0-vortex** [comment]
Actually other way around, shallow, mount and render provided by `enzyme` helps us test React components; it needs a testing library for that to work. Removing this is silly, we should just do better TDD/BDD. For the steps, a contributor has option to run all tests, if tests are failing that should be caused by poor code.

**Reply by mtfoley** [reply]
I think I follow. I had been reading something in testing-library docs implying that it was meant to replace enzyme but the FAQs that talk about snapshots amount to “don’t use snapshots” in my reading

**Reply by 0-vortex** [reply]
> I think I follow. I had been reading something in testing-library docs implying that it was meant to replace enzyme but the FAQs that talk about snapshots amount to “don’t use snapshots” in my reading

From a code coverage perspective we can make storybooks pass through jest and have it better but we'd need to standardize our testing patterns - it might make sense to ditch enzyme for that purpose, while also implementing snapshot testing in storybook. I'm up for doing that because we have close to no "real" TDD and everything is concentrated in storybook. Could just stick to what we're good at/used to.

How does this sound?
